code_generator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -127,6 +128,16 @@ namespace <#= qualifiedNamespace #>
                 view.SetComponentObject(entity, <#= componentDetails.CamelCaseTypeName #>);
 <# } #>
                 view.AddComponent(entity, new NotAuthoritative<<#= componentDetails.TypeName #>>());
+
+                if (view.HasComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity))
+                {
+                    view.RemoveComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity);
+                }
+                else if (!view.HasComponent<ComponentAdded<<#= componentDetails.TypeName #>>>(entity))
```

Comment:
If this is false something has gone wrong. Most likely someone added the component manually.

Reply:
Added an error log message.